Exonoesis/Helping-Hand
# Movement redesign

## What is the purpose of these changes?
To better facilitate the player interaction with entities several changes need to be made to the movement system. There were also several areas that needed to be generalized in preparation for future milestones.

## What has changed?
- Animation logic has been isolated from player movement logic and generalized for any entity capable of animating
- Movement logic has also been generalized for any entity capable of moving 
- Player movement function is now reliant on change detection instead of event broadcasts
- Config file was added to improve Windows build times
- Added a new type of entity, "Interactable," which when collided with and the designated key is pressed will display text to the console

## How can I verify this works?
1.) Run the program and press play on the Main Menu to enter the testing level.
2.) Move the character using the WASD keys and see that the movement is behaving as expected.
3.) Witness the blueberry in the lower left corner of the map. 
4.) Approach it and collide with its blue roundness. 
5.) While touching the blueberry press the "E" key.
6.) Notice text print to the console.
7.) Move away from the blueberry and press "E" again.
8.) Note that nothing occurs when pressing the "E" key while not in contact with the blueberry.

~~~

Closes Issue #20  - Basic Interaction w/ Objects
